### ISSUE TITLE
Explore: a better way to add content on top of images?

### ISSUE THREAD

**Issue by LauraAubin** [LauraAubin/World-Of-Workflow#12]
### TLDR of issues
- Hardcoded position of the overlay and subsequent overlay elements
- Hardcoded size of overlay
- Overlay spanning the rest of the parent container, stretching past the image (creates bugs such as a button is clickable outside the max size of the image)

### Description

Currently, we are creating an overlay with position absolute to sit on top of an image. Then content is added to the overlay, and manually positioned. This feels tedious and spaghetti like. For example, the parent image cannot be resized without having to reposition all of the children elements.

For instance, this is a flat png of a quest. There are empty spaces for a profile, exit button, content writing, and buttons. With only a png, we cannot know where all of these go without manually positioning them.

![image](https://user-images.githubusercontent.com/22782157/62232555-876b5200-b394-11e9-9bf6-8a4fec2d13a1.png)

Another issue with this is that the overlay is not always the same size as the image. The image has a set height/width, but the overlay may span the rest of the area, creating bugs such as the button having extra clickable space around the button image itself.

### Possible solution?

Create an image overlay component that either determines, or is told where available slots live on the image. These positions scale with the size of the image itself, so they are most likely **percentages**.

Then you can set what elements belong to each empty slot. For example,

```html
<Image content={SomeImage} overlay={
   <Overlay.Slot x={50} y={50} width={80} height={50}>
      <div>Some random text</div>
   </Overlay.Slot>

   <Overlay.Slot x={10} y={80} width={100} height={5}>
      <button>Close</button>
   </Overlay.Slot>
}/>
```
